cmd/flow-db.go

```diff
@@ -0,0 +1,223 @@
+package cmd
+
+import (
+	"database/sql"
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/utils"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/write/grpc"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/write/grpc/genericmap"
+
+	// need to import the sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/spf13/cobra"
+)
+
+var flowsDBCmd = &cobra.Command{
+	Use:   "get-flows-db",
+	Short: "",
+	Long:  "",
+	Run:   runFlowDBQuery,
+}
+
+const flowsDB = "/tmp/flows.db"
+
+func runFlowDBQuery(_ *cobra.Command, _ []string) {
+	wg := sync.WaitGroup{}
+	wg.Add(len(ports))
+	for i := range ports {
+		go func(idx int) {
+			defer wg.Done()
+			queryFlowDB(ports[idx])
+		}(i)
+	}
+	wg.Wait()
+}
+
+func queryFlowDB(port int) {
+	os.Remove(flowsDB) // I delete the file to avoid duplicated records.
+	// SQLite is a file based database.
+
+	flowPackets := make(chan *genericmap.Flow, 100)
+	collector, err := grpc.StartCollector(port, flowPackets)
+	if err != nil {
+		log.Error("StartCollector failed:", err.Error())
+		log.Fatal(err)
+	}
+	go func() {
+		<-utils.ExitChannel()
+		close(flowPackets)
+		collector.Close()
+	}()
+
+	log.Println("Creating flows.db...")
+	file, err := os.Create(flowsDB) // Create SQLite file
+	if err != nil {
+		log.Errorf("Failed to create flows db file: %v", err.Error())
+		log.Fatal(err)
+	}
+	file.Close()
+	log.Println("flows.db created")
+	// Open SQLite database
+	db, err := sql.Open("sqlite3", flowsDB)
+	if err != nil {
+		log.Errorf("Error opening database: %v", err.Error())
+		return
+	}
+	defer db.Close()
+
+	// Create messages table if not exists
+	err = createFlowsDBTable(db)
+	if err != nil {
+		log.Errorf("Error creating table: %v", err.Error())
+		log.Fatal(err)
+	}
+	for fp := range flowPackets {
+		handleFlowsDBConnection(db, fp.GenericMap.Value)
+	}
+}
+
+func createFlowsDBTable(db *sql.DB) error {
+	createFlowsTableSQL := `CREATE TABLE flow (
+		"DnsErrno" INTEGER,
+		"Dscp" INTEGER,
+		"DstAddr" TEXT ,
+		"DstPort" INTEGER,
+		"Interface" TEXT,
+		"Proto" INTEGER,
+		"SrcAddr" TEXT,
+		"SrcPort" INTEGER,
+		"Bytes" INTEGER,
+		"Packets" INTEGER,
+		"PktDropLatestDropCause" TEXT,
+		"PktDropBytes" INTEGER,
+		"PktDropPackets" INTEGER,
+		"DnsId" INTEGER,
+		"DnsFlagsResponseCode" TEXT,
+		"DnsLatencyMs" TIMESTAMP,
+		"TimeFlowRTTNs" TIMESTAMP
+	  );` // SQL Statement for Create Table
```

Comment:
Is it on purpose to skip enriched fields here ?
Zone / Host / Namespace / Owner etc

Reply:
Maybe we could have an util function for that to centralize with table display ?

I have in mind in future to reuse frontend config here to generate these:
https://issues.redhat.com/browse/NETOBSERV-1580

Reply:
Also, we should take a look at indexes
Addr, Port, DropState, DropCause, DnsId, DnsErrno & DNSRCode

Reply:
yeah I am focused on the main or key col but we can add as needed we probably can collaborate over this in following PRs

Reply:
sure let's do that in followups